src/upgradeConfig.js

```diff
@@ -61,14 +61,15 @@ function changePresets(config, options = {}) {
     }
 
     if (newPlugins.length > 0) {
-      config.plugins = (config.plugins || []).concat(...newPlugins);
+      config.plugins = (config.plugins || []);
```

Comment:
Since we are not `concat`ing here anymore, this if statement should be moved where we actually need to guard against `config.plugins` from being `undefined`.